src/meta/processors/jobMan/AdminJobProcessor.cpp

```diff
@@ -44,6 +38,32 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
                 break;
             }
 
+            if (cmd == meta::cpp2::AdminCmd::REBUILD_TAG_INDEX ||
+                cmd == meta::cpp2::AdminCmd::REBUILD_EDGE_INDEX) {
+                DCHECK_EQ(paras.size(), 2);
+                auto spaceName = paras[1];
+                auto ret = getSpaceId(spaceName);
+                if (!ret.ok()) {
+                    LOG(ERROR) << "Space " << spaceName << " not found";
+                    errorCode = cpp2::ErrorCode::E_NOT_FOUND;
+                    break;
+                }
+                auto spaceID = ret.value();
+                auto indexName = paras[0];
+                ret = getIndexID(spaceID, indexName);
+                if (!ret.ok()) {
+                    LOG(ERROR) << " Index Name: " << indexName
+                               << " not found" << "in space " << spaceName;
+                    errorCode = cpp2::ErrorCode::E_NOT_FOUND;
+                    break;
+                }
+            }
+
+            auto jobId = autoIncrementId();
+            if (!nebula::ok(jobId)) {
+                errorCode = nebula::error(jobId);
+                break;
+            }
```

Comment:
Let's think about it a little bit more, can we put this logic here?

Reply:
I can't find any other place to put this logic, because a new job is assigned to a new job here. And adding this logic in JobManager is too late...